### ISSUE TITLE
SearchBox A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#7500]
Tracking all of the issues brought up with the 12/28 accessibility pass on SearchBox control 
Details on individual bugs can be found on [SharePoint ](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B4fe91e3e-d70d-4285-b60e-79524a3e8251%7D&action=default&uid=%7B4FE91E3E-D70D-4285-B60E-79524A3E8251%7D&ListItemId=1640&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)
  
 - [x] Bug1: [Accessibility][Functional]: User is not able to search any data in search box as it is not working as per it's functionality.  
- [x] Bug2: [Accessibility][Usable]: After clearing the text written in 'Default Search box', narrator scan mode focus gets stuck while moving forward using 'Down' arrow  key. 
- [x] Bug3: [Accessibility][Usable]: Narrator scan mode focus indicator is not visible on the 'Clear text' button of any 'search' edit field. 
- [ ] Bug4: [Accessibility][Supporting the platform]: Both 'Disable' and ' Enable' Search boxes are visible in same color in high contrast mode. 
tracked in #10624
 
 
Component: SearchBox 
@micahgodbolt

**Comment by micahgodbolt** [comment]
bug 1 is not really a bug. this 'search' control is only the UI to submit a search query. The application is required to actually wire up the search and return results.

**Comment by micahgodbolt** [comment]
Scan mode only highlights the text of a button. Since icon buttons have no text you won't get any visual indicator in scan mode. This repros on the icon button page too. If this is actually a requirement (are visual indicators required?), then it'd be something to resolve in the icon button.

**Comment by micahgodbolt** [comment]
yeah, seems there isn't a color difference for disabled in high contrast.

![image](https://user-images.githubusercontent.com/1434956/50619589-d3adec80-0eae-11e9-8688-4bf54a39af3e.png)

**Comment by micahgodbolt** [comment]
@betrue-final-final do you have designs for disabled search in high contrast. we simply don't have anything specified, and the disabled border/bg show up the same color as the enabled ones.

**Comment by TestAccessibilityGIT** [comment]
For Bug 3, though it's a usability issue but if focus gets visible on icon button that will enhance the usability experience.

**Comment by micahgodbolt** [comment]
i agree that it'd be useful, but even so, this is an icon button issue, and trying to fix it in one place won't solve the systemic problem. we also need to solve it for command bar, and any other control that uses icon buttons

**Comment by micahgodbolt** [comment]
@TestAccessibilityGIT turns out that high contrast mode doesn't differentiate between enabled and disabled inputs
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/1434956/65469496-80a10e80-de1c-11e9-8b02-f1a37904eac2.png)

Is this a high contrast bug, not Fabric bug?

**Comment by 999-Dev** [comment]
Bug3: [Accessibility][Usable]: Narrator scan mode focus indicator is not visible on the 'Clear text' button of any 'search' edit field.
We verify those bugs in below Test Enviourment and found issue is fixed.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18990.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/searchbox
Edge Browser Version:
Microsoft Edge: 44.18990.1.0
Microsoft EdgeHTML: 18.18990
Chrome Browser Version:
Chrome Version: 76.0.3809.132

**Comment by 999-Dev** [comment]
> bug 1 is not really a bug. this 'search' control is only the UI to submit a search query. The application is required to actually wire up the search and return results.

We verify those bugs in below Test Enviourment.

Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18990.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/searchbox
Edge Browser Version:
Microsoft Edge: 44.18990.1.0
Microsoft EdgeHTML: 18.18990
Chrome Browser Version:
Chrome Version: 76.0.3809.132

As per comment closing the bug.

**Comment by micahgodbolt** [comment]
bug 2 is not actually a bug. As per Amit, when narrator enters an input (as it does as soon as you clear the search) scan mode is turned off. SO user would be expected to start typing, or turn scan mode back on and continue.

**Comment by 999-Dev** [comment]
> bug 2 is not actually a bug. As per Amit, when narrator enters an input (as it does as soon as you clear the search) scan mode is turned off. SO user would be expected to start typing, or turn scan mode back on and continue.

Agreed with the comment. Hence closing the bug.
